File: PrimeiroAppJava/app/src/main/java/com/paulo/primeiroappjava/MainActivity.java
package com.paulo.primeiroappjava;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.paulo.primeiroappjava.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
       binding = ActivityMainBinding.inflate(getLayoutInflater());
       setContentView(binding.getRoot());

       binding.btNome.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View v) {
               String teste = binding.edtNome.getText().toString();

               binding.txtNome.setText("Olá :"+teste);
           }
       });

    }
}

File: PrimeiraApp/app/src/main/java/com/paulo/primeiraapp/MainActivity.kt
package com.paulo.primeiraapp

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.paulo.primeiraapp.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    //Variavel binding que vai pertir a ligação dos eleentos da view xml com a classe activity
  private lateinit var binding: ActivityMainBinding


    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //instanticação da variavel biding passando todo o conteudo para ser inflado
        binding = ActivityMainBinding.inflate(layoutInflater)
        //setcontent agora pega o root do binding
        setContentView(binding.root)

        //elemetos
        binding.btnOla.setOnClickListener{
            val nome : String = binding.edtNome.text.toString();
         //   binding.textView.setText("Ola: "+nome);
            binding.textView.setText("Olá ${nome.toUpperCase()}");

        }
    }
}